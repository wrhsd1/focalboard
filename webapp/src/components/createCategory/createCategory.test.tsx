// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react'

import {render} from "@testing-library/react"

import userEvent from "@testing-library/user-event"

import {wrapIntl} from "../../testUtils"

import CreateCategory from "./createCategory"

describe('components/createCategory/CreateCategory', () => {
    it('base case should match snapshot', () => {
        const component = wrapIntl(
            <CreateCategory
                onClose={jest.fn()}
                onCreate={jest.fn()}
                title={
                    <span>{'title'}</span>
                }
            />
        )

        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    it('should call onClose on being closed', () => {
        const onCloseHandler = jest.fn()
        const component = wrapIntl(
            <CreateCategory
                onClose={onCloseHandler}
                onCreate={jest.fn()}
                title={
                    <span>{'title'}</span>
                }
            />
        )

        const {container} = render(component)
        const cancelBtn = container.querySelector('.createCategoryActions > .Button.danger')
        expect(cancelBtn).toBeTruthy()
        userEvent.click(cancelBtn as Element)
        expect(onCloseHandler).toBeCalledTimes(1)

        const closeBtn = container.querySelector('.toolbar > .IconButton')
        expect(closeBtn).toBeTruthy()
        userEvent.click(closeBtn as Element)
        expect(onCloseHandler).toBeCalledTimes(2)
    })

    it('should call onCreate on pressing enter', () => {
        const onCreateHandler = jest.fn()
        const component = wrapIntl(
            <CreateCategory
                onClose={jest.fn()}
                onCreate={onCreateHandler}
                title={
                    <span>{'title'}</span>
                }
            />
        )

        const {container} = render(component)
        const inputField = container.querySelector('.categoryNameInput')
        expect(inputField).toBeTruthy()
        userEvent.type(inputField as Element, 'category name{enter}')
        expect(onCreateHandler).toBeCalledWith('category name')
    })

    it('should show initial value', () => {
        const onCreateHandler = jest.fn()
        const component = wrapIntl(
            <CreateCategory
                initialValue='Dwight prank ideas'
                onClose={jest.fn()}
                onCreate={onCreateHandler}
                title={
                    <span>{'title'}</span>
                }
            />
        )

        const {container} = render(component)
        const inputField = container.querySelector('.categoryNameInput')
        expect(inputField).toBeTruthy()
        expect((inputField as HTMLInputElement).value).toBe('Dwight prank ideas')
    })

    it('should clear input field on clicking clear icon', () => {
        const onCreateHandler = jest.fn()
        const component = wrapIntl(
            <CreateCategory
                initialValue='Dunder Mifflin'
                onClose={jest.fn()}
                onCreate={onCreateHandler}
                title={
                    <span>{'title'}</span>
                }
            />
        )

        const {container} = render(component)
        const inputField = container.querySelector('.categoryNameInput')
        expect(inputField).toBeTruthy()
        expect((inputField as HTMLInputElement).value).toBe('Dunder Mifflin')

        const clearBtn = container.querySelector('.clearBtn')
        expect(clearBtn).toBeTruthy()
        userEvent.click(clearBtn as Element)
        expect((inputField as HTMLInputElement).value).toBe('')
    })
})
