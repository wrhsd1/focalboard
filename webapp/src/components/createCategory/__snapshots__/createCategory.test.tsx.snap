// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/createCategory/CreateCategory base case should match snapshot 1`] = `
<div>
  <div
    class="Dialog dialog-back CreateCategoryModal"
  >
    <div
      class="backdrop"
    />
    <div
      class="wrapper"
    >
      <div
        class="dialog"
        role="dialog"
      >
        <div
          class="toolbar"
        >
          <button
            aria-label="Close dialog"
            class="IconButton size--medium"
            title="Close dialog"
            type="button"
          >
            <i
              class="CompassIcon icon-close CloseIcon"
            />
          </button>
          <div
            class="toolbar--right"
          />
        </div>
        <div
          class="CreateCategory"
        >
          <h3>
            <span>
              title
            </span>
          </h3>
          <div
            class="inputWrapper"
          >
            <input
              class="categoryNameInput"
              maxlength="100"
              placeholder="Name your category"
              type="text"
              value=""
            />
          </div>
          <div
            class="createCategoryActions"
          >
            <button
              class="Button danger size--medium"
              type="button"
            >
              <span>
                Cancel
              </span>
            </button>
            <button
              class="Button size--medium"
              disabled=""
              type="button"
            >
              <span>
                Create
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
